### PR TITLE
ascend add range op

### PR DESCRIPTION
**Pull request by OleNet**
### PR types

New features

### PR changes

OPs

### Describe

[NPU] add npu kernel for range op

### CODE REVIEW

**Comment by OleNet** [comment]
image
image
image

File: paddle/fluid/operators/range_op_npu.cc [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
+
```
Comment (liym27) [inline paddle/fluid/operators/range_op_npu.cc:2]: 2016 -> 2021

File: paddle/fluid/operators/range_op_npu.cc [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+#include "paddle/fluid/framework/operator.h"
+#include "paddle/fluid/framework/program_desc.h"
+#include "paddle/fluid/operators/dropout_op.h"
+#include "paddle/fluid/operators/math/math_function.h"
+
+namespace paddle {
+namespace operators {
+
+template <typename T>
+void GetSizeNPURange(T start, T end, T step, int64_t* size) {
```
Comment (liym27) [inline paddle/fluid/operators/range_op_npu.cc]: This function is the same as `GetSize` in file "range_op.h".  
To better maintain the code, please delete `GetSizeNPURange` and call `GetSize` instead.

File: paddle/fluid/operators/range_op_npu.cc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    std::vector<T> odata;
+    T value = start;
+    for (int64_t i = 0; i < size; ++i) {
+      odata.push_back(value);
+      value += step;
+    }
+
+    framework::TensorFromVector(odata, context.device_context(), out);
+  }
```
Comment (liym27) [inline paddle/fluid/operators/range_op_npu.cc:64]: Why not use npu op `Range` directly for performance optimization?

**Review by liym27 — CHANGES_REQUESTED** [review]

**Comment by liym27** [comment]
Please specify PR description.
You can refer to this https://github.com/PaddlePaddle/Paddle/pull/31275

**Pull request merged**